File: cash.c
// import libraries
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // create variables
    float amount; // amount is the amount of change entered by the user
    int cents = 0; // cents is the amount of change in cents
    int count = 0; // count is the number of coins needed
    int amount_left = 0; // amount_left is the amount of change left

    // this do-while loops if the user puts in invalid input
    do
    {
        amount = get_float("Change owed: ");
    }
    while (amount <= 0); // ensures input is positive

    cents = (int)round(amount * 100);

    amount_left = cents;
    count += amount_left / 25;  // this counts the number of quarters
    amount_left %= 25;

    count += amount_left / 10;  // this counts the number of dimes
    amount_left %= 10;

    count += amount_left / 5;  // this counts the number of nickels
    amount_left %= 5;

    count += amount_left; // this counts the number of pennies

    // display the count of coins
    printf("You get %d coins\n", count);
}

File: PSET6/cash.py
# import libraries
from cs50 import get_float
from math import floor

cents = 0
count = 0
amount_left = 0

# ask the user to input their change
amount = get_float("Changed owed: ")

# ensures that the value is positive
while amount <= 0:
    amount = get_float("Changed owed: ")

# convert dollars to cents
cents = floor(amount * 100)

# counts the number of quarters
amount_left = cents
count += floor(amount_left / 25)
amount_left %= 25

# counts the number of dimes
count += floor(amount_left / 10)
amount_left %= 10

# counts the number of nickes
count += floor(amount_left / 5)
amount_left %= 5

# counts the number of pennies
count += floor(amount_left)

print(f"You get {int(count)} coins")
